### ISSUE TITLE
DatePicker pick wrong date when custom mask is applied

### ISSUE THREAD

**Issue by prophe05** [quasarframework/quasar#6610]
**Describe the bug**
DatePicker pick wrong date when custom mask (MMM, YYYY) is used with default-view="Months" and emit-immediately property.

**Codepen/jsFiddle/Codesandbox (required)**
https://codepen.io/prophe05/pen/XWbEzGw?editable=true&editors=101

**To Reproduce**
1. Open codepen example
2. select the month from the picker (should pick the right date)
3. try to pick another month (should pick the previous one, e.g Feb clicked, should pick Jan)

**Expected behavior**
The right month should be selected

**Comment by mateonunez** [comment]
When you use a mask with day reference like"MM-YYYY" in `date<Object>` day prop is initialized with `1`.

Casting this object this is you should return
```js 
defaultDate: { year: 2020, month: 3, day: 1 } 
```
```js 
new Date(defaultDate)
```
```
Mon Jun 01 2020 00:00:00 GMT+0200 (Central European Summer Time)
```
But if ```day: 0```

```
Sun May 31 2020 00:00:00 GMT+0200 (Central European Summer Time)
```
Infact, Quasar substract 1 from the month when parse date in ```__updateValue (date, reason)```

```js
const val = this.calendar === 'persian'
  ? date.year + '/' + pad(date.month) + '/' + pad(date.day)
  : formatDate(
      new Date(
        date.year,
        date.month - 1,
        date.day,
        this.extModel.hour,
        this.extModel.minute,
        this.extModel.second,
        this.extModel.millisecond
     ),
   this.mask,
  this.computedLocale,
  date.year
  )
```

A work around is checking when day is selected or when it isn't.

```js
date.month = date.day ? date.month - 1 : date.month
```

Byeee!

**Comment by rstoenescu** [comment]
Fix will be available in "quasar" v1.9.11.
